### ISSUE TITLE
Add tooling in support of auto-update of code excerpts in docs

### ISSUE THREAD

**Issue by chalin** [open-telemetry/opentelemetry.io#1635]
@svrnm - I remember you mentioning this somewhere, but I can't find the reference. Anyhow, I'm creating this issue to track progress on this work. /cc @cartermp 

At the start of the year I introduced a simple verbatim code-copy shortcode, but what I have in mind would be more flexible.

- [x] #3948
- [ ] Replace Dart-based tooling with JS/Node-based versions
  - [ ] Investigate JS-based tooling feature set. Might it be a drop-in replacement? [Add a reference here to the JS-based tooling.]

Tooling use:

- [x] #4642

**Comment by svrnm** [comment]
we discussed that on slack and/or the comms call a few months back, but I can't find it either.

Curious to see & discuss what you have in mind

**Comment by svrnm** [comment]
We discussed this during yesterday's SIG meeting. @chalin will provide a proposal following some best practices other projects have implemented today already. For testing the [go/example/dice](https://github.com/open-telemetry/opentelemetry-go/tree/main/example/dice?rgh-link-date=2023-12-04T19%3A01%3A34Z) and the [go getting started](https://opentelemetry.io/docs/instrumentation/go/getting-started/) will be used (cc @pellared, who has contributed both)

@ekquasar the PR may or may not answer your question around keeping things in sync, so let's wait for what @chalin's prototype will look like, early next year.

**Comment by zeitlinger** [comment]
I'm in the process of adding more snippets, so this feature would be great.

See https://github.com/open-telemetry/opentelemetry.io/pull/4642

**Comment by svrnm** [comment]
@zeitlinger we have tooling place, see https://github.com/open-telemetry/opentelemetry.io/pull/3948 and some of the Go Docs related to that, it's still under observation, but you can take a look and see if you can make use of it for java as well

**Comment by zeitlinger** [comment]
why are the go samples not up-to-date then?

**Comment by svrnm** [comment]
> why are the go samples not up-to-date then?

they aren't? I guess because it is still a manual process to update them (cc @chalin)

**Comment by chalin** [comment]
Yes, given that this is still in development (and I'm hoping to bring in simpler tooling as soon as I have time to devote to this), the process is still manual.

**Comment by chalin** [comment]
Let me refresh those code excerpts now.

**Comment by svrnm** [comment]
> Let me refresh those code excerpts now.

https://github.com/open-telemetry/opentelemetry.io/pull/4678?notification_referrer_id=NT_kwDOABcwjbMxMTA3NzQxNDgwNDoxNTE5NzU3#issuecomment-2165917376 I gave it a try 🙈 , since some of the code has been put into a separate file this is not only an update but also needs some textual changes

**Comment by chalin** [comment]
Thanks @svrnm. I'll let you handle it from here.

**Comment by chalin** [comment]
@jack-berg - I'm hoping to have a new NPM-package based version of the tooling available soon. I see that you're working on PRs that use code excerpts. Will we need to coordindate for the tooling update? I'm hoping that it will be a drop-in replacement.

**Comment by jack-berg** [comment]
> Will we need to coordindate for the tooling update? 

I assume you'll be updating all the existing excerpts to use whatever syntax changes are required? If so, and if your PR is merged first, I'm happy to update the excerpts my PR uses to and follow the pattern.

**Comment by chalin** [comment]
> I assume you'll be updating all the existing excerpts to use whatever syntax changes are required?

FYI: for this first JS/TS version, there will be no syntax changes. It's mainly the tooling setup that will change (to use NPM), and we'll be dropping the need Dart, etc.

**Comment by chalin** [comment]
@jack-berg et all: the new tool is ready. Here's a PR with the tool switch and some cleanup:

- #9638
